inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/instrumentation/ClassInstrumentationConfiguration.java

```diff
@@ -0,0 +1,70 @@
+package rocks.inspectit.oce.core.instrumentation;
+
+import lombok.Value;
+import net.bytebuddy.description.type.TypeDescription;
+import rocks.inspectit.oce.core.config.model.instrumentation.InstrumentationSettings;
+import rocks.inspectit.oce.core.instrumentation.special.SpecialSensor;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Defines the actual or requested state of the instrumentation of a single class.
+ */
+@Value
+class ClassInstrumentationConfiguration {
+
+    /**
+     * The configuration representing that no instrumentation of the class is performed.
+     */
+    static final ClassInstrumentationConfiguration NO_INSTRUMENTATION = new ClassInstrumentationConfiguration(Collections.emptySet(), null);
+
+    /**
+     * The sensors which are active for the target class.
+     */
+    Set<SpecialSensor> activeSensors;
+
+    /**
+     * The inspectIT configuration used to derive this configuration object.
+     * Can be null, but only if no sensors active.
+     */
+    InstrumentationSettings activeConfiguration;
+
+    /**
+     * Compares this instrumentation configuration against another.
+     * Two instrumentations are considered to be the "same" if they result in the same bytecode changes.
+     *
+     * @param forType the type for which this configuration is meant.
+     * @param other   the other configuration to compare against
+     * @return true if both are the same, false otherwise
+     */
+    public boolean isSameAs(TypeDescription forType, ClassInstrumentationConfiguration other) {
+        if (!activeSensors.equals(other.activeSensors)) {
```

Comment:
Here it's better to go for `Collections.disjoint` where you can find out if elements of in the collections are same or not.. this way you don't depend on the collection implementation to be same...

Reply:
As we create the set ourselves, equals is fine and faster.